objectcache/objectcache_interface.go

```diff
@@ -0,0 +1,69 @@
+package objectcache
+
+import (
+	"context"
+
+	"github.com/kubescape/k8s-interface/k8sinterface"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type KubernetesCache interface {
```

Comment:
Why do we need this one?

Reply:
Amit wanted me to create generic interface because ProcessEvent gets generic interfaces